docs/api/io/BufferedSerial.md

```diff
@@ -0,0 +1,71 @@
+# BufferedSerial
+
+<span class="images">![](https://os.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_buffered_serial.png)<span>`BufferedSerial` class hierarchy</span></span>
+
+The `BufferedSerial` class provides UART functionality, it is the recommended class for serial data transfers. It allows sending and receiving bytes of data in a sequence using separate transmit (TX) and receive pins (RX). Communication can be between two processors or for sending text to a console.
+
+TX and RX pins can be specified as Not Connected (NC) for simplex (unidirectional) communication or as valid pins for full duplex (bi-directional) communication.
+
+Data is sent/received at a pre-defined speed called baud rate. Standard baud rate include 9600, 119200, 115200 or others.
+
+Pieces of data are transmitted using data packets of configurable sizes divided in different sections which include:
+
+* Start bit: Indicates the start of UART data transmission
+* Data frame: Can be 5 to 8 (or 9 if a parity bit is not used) bits long for the actual data being transferred.
+* Parity bit: Optional bit, used for data error detection.
+* Stop bits: Can be 1 to 2 bits long to signal the end of a data packet.
+
+The `BufferedSerial` calls the underlying HAL API functions. See the [porting guide](../porting/serial-port.html) for target serial support.
+
+When the RX interrupt is trigged, the `BufferedSerial` class stores the byte(s) available to read from the hardware buffer to an internal intermediary buffer. When a read request is made, the `BufferedSerial` class uses a mutex lock and enters a critical section to read out the number of bytes requested if as many are available in the intermediary buffer.
+
+To transmit multiple bytes, the class uses an intermediary buffer to store the bytes to send and monitors the serial interface to transfer them to the hardware buffer as soon as it is available. However, all bytes are written unbuffered if in a critical section.
+Using intermediary buffers allows it to be used reliably for input from non-interrupt context whilst avoiding excess spinning waiting for transmission buffer space.
+
+The RX and TX buffers are circular buffers with pre-allocated sizes configurable using [`drivers/mbed_lib.json`](https://github.com/ARMmbed/mbed-os/blob/master/drivers/mbed_lib.json).
+
+## Configuration
+
+The following parameters can be configured at object instantiation:
+
+  - _TX_
+  - _RX_
+  - _Baud rate_ - defaults to the value configured in `mbed-os/platform/mbed_lib.json`.
```

Comment:
It would be nicer to write the default as well (rather than asking the user to check it in a file). In any case you mentioni it line 42.

Reply:
If we happen to change them we will have to modify this also.

Reply:
There is no default for TX or RX. There is a default for the rest and I mention it below. I do not see what to add here.